trl/import_utils.py

```diff
@@ -14,28 +14,38 @@
 import importlib
 import os
 import sys
-from importlib.util import find_spec
 from itertools import chain
 from types import ModuleType
 from typing import Any
 
+from transformers.utils.import_utils import _is_package_available
+
 
 if sys.version_info < (3, 8):
     _is_python_greater_3_8 = False
 else:
     _is_python_greater_3_8 = True
 
-
-def is_peft_available() -> bool:
-    return find_spec("peft") is not None
+_diffusers_available = _is_package_available("diffusers")
```

Comment:
Why don't we just call `_is_package_available()` within the methods below instead of defining these private variables?

Reply:
This is how it's done in `transformers`.  I've opted for consistency.

https://github.com/huggingface/transformers/blob/main/src/transformers/utils/import_utils.py#L97

I don't know why they do that. Especially since it involves something ugly like this:

https://github.com/huggingface/trl/blob/f8811b6b4af81479909381b96ca9d879225efb9e/tests/test_no_peft.py#L78

Reply:
I see, ok let's follow their approach then - could you just add a comment which points to their convention for future us / maintainers?

Reply:
See #2073